src/lib/base/buf_input.h

```diff
@@ -0,0 +1,73 @@
+/*
+* Buffered Input
+* (C) 2023 Jack Lloyd
+*     2023 Fabian Albert, René Meusel - Rohde & Schwarz Cybersecurity
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_BUFFERED_INPUT_H_
+#define BOTAN_BUFFERED_INPUT_H_
+
+#include <botan/mem_ops.h>
+
+#include <span>
+#include <string_view>
+
+namespace Botan {
+
+/**
+* This class represents any kind of computation which uses an internal
+* state, such as hash functions, MACs or XOFs and allows filling that
+* internal state with an arbitrary number of streamlined update() calls.
+*/
+class BOTAN_PUBLIC_API(3, 2) Buffered_Input {
+   public:
+      /**
+      * Add new input to process.
+      * @param in the input to process as a byte array
+      * @param length of param in in bytes
+      */
+      void update(const uint8_t in[], size_t length) { add_data(in, length); }
```

Comment:
`Buffered_Computation` is already deprecated, I don't think we should use it for the XOF

This also allows some modernization and minimization of the API. In fact we could literally go to just taking a `std::span<const uint8_t>` and be done with it.

Reply:
Mhm, fair enough. Albeit, I found it quite handy to have `update_le()`, for instance.

You're proposing to omit the common base interface altogether, right?

Reply:
Exactly so, yes